components/brave_wallet/browser/network_manager_unittest.cc

```diff
@@ -0,0 +1,1050 @@
+
+// DEPRECATED 01/2024. For migration only.
+std::string GetSolanaSubdomainForKnownChainId(const std::string& chain_id);
+std::string GetFilecoinSubdomainForKnownChainId(const std::string& chain_id);
+std::string GetBitcoinSubdomainForKnownChainId(const std::string& chain_id);
+std::string GetZCashSubdomainForKnownChainId(const std::string& chain_id);
+std::string GetKnownEthNetworkId(const std::string& chain_id);
+std::string GetKnownSolNetworkId(const std::string& chain_id);
+std::string GetKnownFilNetworkId(const std::string& chain_id);
+std::string GetKnownBtcNetworkId(const std::string& chain_id);
+std::string GetKnownZecNetworkId(const std::string& chain_id);
+
+namespace {
+
+// void UpdateCustomNetworks(PrefService* prefs,
+//                           const std::vector<base::Value::Dict>& values,
+//                           brave_wallet::mojom::CoinType coin) {
+//   ScopedDictPrefUpdate update(prefs, kBraveWalletCustomNetworks);
+//   base::Value::List* list = update->EnsureList(GetPrefKeyForCoinType(coin));
+//   list->clear();
+//   for (auto& it : values) {
+//     list->Append(it.Clone());
+//   }
+// }
+
+// void UpdateCustomNetworks(PrefService* prefs,
+//                           const base::Value::Dict& value,
+//                           brave_wallet::mojom::CoinType coin) {
+//   std::vector<base::Value::Dict> values;
+//   values.push_back(value.Clone());
+//   UpdateCustomNetworks(prefs, values, coin);
+// }
+
+// std::vector<brave_wallet::mojom::NetworkInfoPtr> GetAllEthCustomChains(
+//     NetworkManager* network_manager) {
+//   return network_manager->GetAllCustomChains(
+//       brave_wallet::mojom::CoinType::ETH);
+// }
+
+}  // namespace
+
+class NetworkManagerUnitTest : public testing::Test {
+ public:
+  NetworkManagerUnitTest() = default;
+  void SetUp() override {
+    RegisterProfilePrefs(prefs_.registry());
+    network_manager_ = std::make_unique<NetworkManager>(&prefs_);
+  }
+
+  PrefService* prefs() { return &prefs_; }
+  NetworkManager* network_manager() { return network_manager_.get(); }
+
+ protected:
+  sync_preferences::TestingPrefServiceSyncable prefs_;
+  std::unique_ptr<NetworkManager> network_manager_;
+};
+
+TEST_F(NetworkManagerUnitTest, GetAllCustomChainsTest) {
+  for (auto coin : kAllCoins) {
+    SCOPED_TRACE(coin);
+    ASSERT_TRUE(network_manager()->GetAllCustomChains(coin).empty());
+    mojom::NetworkInfo chain1 = GetTestNetworkInfo1("chain1", coin);
+    mojom::NetworkInfo chain2 = GetTestNetworkInfo2("chain2", coin);
+    network_manager()->AddCustomNetwork(chain1);
+    network_manager()->AddCustomNetwork(chain2);
+
+    // std::vector<base::Value::Dict> values;
+
+    // mojom::NetworkInfo chain1 = GetTestNetworkInfo1("chain1", coin);
+    // values.push_back(NetworkInfoToValue(chain1));
+
+    // mojom::NetworkInfo chain2 = GetTestNetworkInfo2("chain2", coin);
+    // values.push_back(NetworkInfoToValue(chain2));
+    // UpdateCustomNetworks(prefs(), std::move(values), coin);
```

Comment:
what are these?

Reply:
cleaned this: https://github.com/brave/brave-core/pull/24365/commits/90f3a806f44fb0020b62971c6a305622a0c6a362